rosidl_generator_py/resource/_msg.py.em

```diff
@@ -288,12 +297,15 @@ if isinstance(type_, AbstractNestedType):
 @[  end if]@
 ,  # noqa: E501
 @[end for]@
+        rosidl_parser.definition.BasicType('boolean'),  # noqa: E501
     )
 
-    def __init__(self, **kwargs):
-        assert all('_' + key in self.__slots__ for key in kwargs.keys()), \
-            'Invalid arguments passed to constructor: %s' % \
-            ', '.join(sorted(k for k in kwargs.keys() if '_' + k not in self.__slots__))
+    def __init__(self, check_fields=False, **kwargs):
+        self._check_fields = check_fields or ros_python_check_fields
```

Comment:
After playing with this for a bit, I actually think we should change up the logic here.

There are currently two problems that I see:
1. Doing `check_fields or ros_python_check_fields` isn't actually correct; if the `ROS_PYTHON_CHECK_FIELDS` environment variable is not defined, `ros_python_check_fields` evaluates to `None`, and if it is defined, it evaluates to a string.
2. If the user explicitly passes `check_fields=False`, then we probably want to honor that, *even if ROS_PYTHON_CHECK_FIELDS was also set to true*.

To solve both of these, I think we need to do the following:

```suggestion
    def __init__(self, **kwargs):
        if 'check_fields' in kwargs:
            self._check_fields = kwargs['check_fields']
        else:
            self._check_fields = ros_python_check_fields is not None and ros_python_check_fields == "1"
```

Reply:
> I think this would make more sense than to leave it as none. Regarding the value the user sets to the variable, I'm a little troubled since the can basically set the variable to anything he wants right? Wouldn't it makes sense to add a check up during the lookup of the variable?
> ```

Yes, they can set it to anything.  But my thinking is that maybe in the future, we want to have more fine-grained checking.  Like if you set the environment variable to "1", it only checks simple types, but if you set it to "2", it also checks array elements.  We don't need to define all of the semantics here, but if we make sure that this is a known value, we can easily expand it in the future.

So with that in mind, I'll suggest we combine your approach and my approach.  Basically when looking up the variable with getenv, we do:

```
from os import getenv
ros_python_check_fields = getenv("ROS_PYTHON_CHECK_FIELDS", default='') 
```

And then when we use it, we do:
```
        if 'check_fields' in kwargs:
            self._check_fields = kwargs['check_fields']
        else:
            self._check_fields = ros_python_check_fields == "1"
```

Does that make sense to you?

Reply:
Ok, yeah it makes sense. I like the idea of  expanding it in the future if needed/wanted.